src/auto-advance.js

```diff
@@ -50,11 +52,14 @@ const setup = (player, delay) => {
   // Before queuing up new auto-advance behavior, check if `seconds` was
   // called with a valid value.
   if (!validSeconds(delay)) {
+    player.playlist.autoadvance_.delay = null;
     return;
   }
 
+  player.playlist.autoadvance_.delay = delay;
```

Comment:
The main change in this file is to set the delay in the `setup` function. This allows us to provide more normalized values and keep ownership of the `autoadvance_` object in this module, which remaining backward-compatible.